test/src/unit-cppapi-group.cc

```diff
@@ -1165,3 +1165,47 @@ TEST_CASE(
     CHECK(group.is_relative(name));
   }
 }
+
+TEST_CASE_METHOD(
+    GroupCPPFx,
+    "C++ API: Group with cycle, dump and is_relative",
+    "[cppapi][group][cycle][non-rest]") {
+  // Create groups that form a cycle
+  std::string group_a_uri = vfs_test_setup_.array_uri("group_a");
+  std::string group_b_uri = vfs_test_setup_.array_uri("group_b");
+  tiledb::Group::create(ctx_, group_a_uri);
+  tiledb::Group::create(ctx_, group_b_uri);
+
+  // Open group A and add group B as a member
+  {
+    tiledb::Group group_a(ctx_, group_a_uri, TILEDB_WRITE);
+    group_a.add_member(group_b_uri, false, "group_b");
+    group_a.close();
+  }
+
+  // Open group B and add group A as a member, creating a cycle
+  {
+    tiledb::Group group_b(ctx_, group_b_uri, TILEDB_WRITE);
+    group_b.add_member(group_a_uri, false, "group_a");
+    group_b.close();
+  }
+
+  // Open group A for reading and test cycle handling
+  tiledb::Group group_a(ctx_, group_a_uri, TILEDB_READ);
+
+  // Test is_relative - this should not hang even with cycles
+  CHECK_NOTHROW(group_a.is_relative("group_b"));
```

Comment:
Was this actually hanging before? I don't see how this PR addresses it

Reply:
It wasn't.